### ISSUE TITLE
`TelescopeNormal` highlight group doesn't color full background using `dropdown` theme

### ISSUE THREAD

**Issue by mrjones2014** [nvim-telescope/telescope.nvim#1596]
### Description

When the results aren't enough to fill the full window when using Telescope's "dropdown" theme, the `TelescopeNormal` highlight group doesn't apply to the space below the results.

I've set the highlight groups like so:

```
  vim.cmd('hi link TelescopeNormal NormalFloat')
  vim.cmd('hi link TelescopeBorder FloatBorder')
  vim.cmd('hi link TelescopePreviewNormal NormalFloat')
  vim.cmd('hi link TelescopeResultsNormal NormalFloat')
```

### Neovim version

```markdown
NVIM v0.6.0
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by brew@HMBRW-A-001-M1-004.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.6.0/share/nvim"

Run :checkhealth for more info
```

### Operating system and version

macOS 12.0.1

### checkhealth telescope

```markdown
telescope: require("telescope.health").check()
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - OK: nvim-treesitter installed.

## Checking external dependencies
  - OK: rg: found ripgrep 13.0.0
  - OK: fd: found fd 8.3.0

## ===== Installed extensions =====

## Telescope Extension: `fzf`
  - INFO: No healthcheck provided

## Telescope Extension: `tldr`

## tldr command
  - OK: configured command "tldr" found and is executable
```

### Steps to reproduce

1. Run `:Telescope find_files theme=dropdown`
1. Search for something that will return a small amount of or zero results
1. The space below the results doesn't match the rest of the window background (see screenshot below)
2. 
![CleanShot-2021-12-14-at-07 24 01](https://user-images.githubusercontent.com/8648891/145997738-9ec96f6a-89af-42c2-adb9-ab23ab995b72.png)

### Expected behavior

The space below the results should have the same background color as the rest of the window.

### Actual behavior

The space below the results has a different background color. I am not sure what highlight group it is using, or how to find that out.

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          { 'nvim-telescope/telescope-fzf-native.nvim', run = 'make' },
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
      "mrjones2014/lighthaus.nvim", -- color scheme that sets highlight groups
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('fzf')
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
  require("lighthaus").setup({ bg_dark = true })
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by l-kershaw** [comment]
I think this might be an issue caused by our implementation of popups over in `plenary.nvim`.
We don't set the `EndOfBuffer` highlight and maybe we should? I'll have a proper look at it later

**Comment by mrjones2014** [comment]
One thing to note, it only seems to happen with the `theme=dropdown`. With the default theme it works just fine, see screenshot:

![CleanShot-2021-12-14-at-07 54 34](https://user-images.githubusercontent.com/8648891/146002206-d96b8923-4738-49d3-a2d0-16c1ca9bfd14.png)
